doc/topics/the-event-loop-timers-and-nexttick.md

```diff
@@ -0,0 +1,375 @@
+this in further detail later in this topic.  
+
+## Event Loop Explained 
+
+When Node.js starts, it initializes the event loop, processes the provided
+input script (or drops into the REPL, which is not covered in this document)
+which may make async API calls, schedule timers, or call `process.nextTick()`, 
+then begins processing the event loop.  
+
+The Following diagram shows a simplified overview of the event loop's
+order of operations.
+
+       ┌───────────────────────┐
+    ┌─>│        timers         │
+    │  └──────────┬────────────┘
+    │  ┌──────────┴────────────┐
+    │  │   pending callbacks   │
+    │  └──────────┬────────────┘      ┌───────────────┐
+    │  ┌──────────┴────────────┐      │   incoming:   │
+    │  │         poll          │<─────│  connections, │
+    │  └──────────┬────────────┘      │   data, etc.  │
+    │  ┌──────────┴────────────┐      └───────────────┘
+    └──│     setImmediate      │
+       └───────────────────────┘
+
+*note: each box will be referred to as a "phase" of the event loop.*
+
+Each phase has a FIFO queue of callbacks to execute. While each phase  is
+special in its own way, generally, when the event loop enters a  given phase,
+it will perform any operations specific to that phase,  then execute callbacks
+in that phase's queue until the queue has been exhausted or the maximum number
+of callbacks have executed.  When the queue has been exhausted or the callback
+limit is reached, the event loop will move to the next  phase, and so on.
+
+Since any of these operations may schedule _more_ operations and new events
+processed in the `poll` phase are queued by the kernel, poll events can be
+queued while polling events are being processed.  As a result, long running
+callbacks can allow the poll phase to run much longer than a timer's
+threshold.  See the [`timers`](#timers) and [`poll`](#poll) sections for more
+details.
+
+_**NOTE:** There is a slight discrepancy between the Windows and the
+Unix/Linux implementation, but that's not important for this demonstration.
+The most important parts are here.  There are actually seven or eight steps,
+but the ones we care about — ones that Node.js actually uses are these four._
+
+
+## Phases Overview:
+
+* `timers`: this phase executes callbacks scheduled by `setTimeout()` 
+  and `setInterval()`.
+* `pending callbacks`: this phase executes callbacks for specific types of TCP
+  errors, for example.
+* `poll`: this is the phase in which the event loop either processes
+  its queue of callbacks, or sits and waits for new callbacks for incoming
+  connections, completion of async file I/O, DNS operations, etc. to be added 
+  to the queue. Ideally, most scripts spend most of their time here. If not, 
+  it might be time consider refactoring or scaling to more processes.
+* `setImmediate`: This phase allows a person to execute callbacks
+  immediately after the `poll` phase has completed. 
```

Comment:
Possibly make a note that when a setImmediate is schedules, the poll phase will still run and check for any waiting notifications. Which will then be processed. If not then execution will proceed immediately to calling the setImmediate.